Assets/Scripts/EnemyController.cs

```diff
@@ -85,6 +88,16 @@ void KnightEnemyBehavior()
         }
         else if (enemyRenderer.isVisible && distanceFromEnemy <= detectDistance)
         {
+            if (playerTransform.position.x > gameObject.transform.position.x && !facingRight)
+            {
+                transform.Rotate(0f, 180f, 0);
+                facingRight = true;
+            }
+            else if (playerTransform.position.x < gameObject.transform.position.x && facingRight)
+            {
+                transform.Rotate(0f, 180f, 0);
+                facingRight = false;
+            }
             playerDetected = true;
```

Comment:
look into flip x on the transform of the gameobject, might be able to simplify your code or outright take it out